libraries/RC_Channel/RC_Channels.cpp

```diff
@@ -234,6 +240,48 @@ uint32_t RC_Channels::enabled_protocols() const
     return uint32_t(_protocols.get());
 }
 
+
+/*
+  check for pilot input on rudder stick for arming/disarming
+*/
+void RC_Channels::rudder_arm_disarm_check()
+{
+    const RC_Channel *channel = rc().get_arming_channel();
+    if (channel == nullptr) {
+        return;
+    }
+
+    if (!has_valid_input() ||
+        abs(channel->get_control_in()) <= 4000) {
+        // not trying to (or no longer trying to) arm or disarm - or
+        // we don't have valid RC input.
+        rudder_arm_timer = 0;
+        return;
+    }
+
+    const uint32_t now = AP_HAL::millis();
+    if (rudder_arm_timer == 0) {
+        // first time we've seen the attempt
+        rudder_arm_timer = now;
+        return;
+    }
+
+    if (now - rudder_arm_timer >= 3000) {
+        // time to try to arm or disarm:
+        if (channel->get_control_in() > 4000) {
```

Comment:
why not reset timer when we call this? this looks like it calls continuously

Reply:
Fixed.

Reply:
doesn't that fix break the 10s check for copter?

Reply:
Yes, it did.  Fixed and tested.